src/__tests__/words.spec.ts

```diff
@@ -0,0 +1,223 @@
+import {describe, it, expect} from 'vitest';
+import {
+  zeroVector,
+  add,
+  sub,
+  neg,
+  norm,
+  scalarMult,
+  rotate,
+  getStepVector,
+  stepSignature,
+  wordArity,
+  stepVectorArity,
+} from '../words';
+
+function clone(x) {
+  return JSON.parse(JSON.stringify(x));
+}
+
+describe('add()', () => {
+  it('does not mutate input', () => {
+    const vector = {a: 1, b: 2, c: -3};
+    const originalVector = clone(vector);
+    add(vector, zeroVector());
+    expect(vector).toStrictEqual(originalVector);
+    add(zeroVector(), vector);
+    expect(vector).toStrictEqual(originalVector);
+    const zero = zeroVector();
+    add(zero, zeroVector());
+    expect(zero).toStrictEqual(zeroVector());
+    add(zeroVector(), zero);
+    expect(zero).toStrictEqual(zeroVector());
+  });
+  it('has identity element zeroVector()', () => {
+    const vector = {a: 1, b: 2, c: -3};
+    expect(add(vector, zeroVector())).toStrictEqual(vector);
+    expect(add(zeroVector(), vector)).toStrictEqual(vector);
+  });
+  it('can update existing components', () => {
+    const v1 = {a: 1, b: 2, c: -3};
+    const v2 = {a: 1, b: 1, c: 1};
+    const v1Clone = clone(v1);
+    const v2Clone = clone(v2);
+    const sum = {a: 2, b: 3, c: -2};
+    expect(add(v1, v2)).toStrictEqual(sum);
+    expect(add(v2Clone, v1Clone)).toStrictEqual(sum);
```

Comment:
Instead of cloning, these tests should now be used to verify that `v1` and `v2` remain unchanged.

Reply:
I added the 'does not mutate input' today. Maybe I should do the same for the other fucntions in words.ts as well.

Reply:
I added the `'does not mutate input'` test for `add()` and `scalarMult()` today. Maybe I should do the same for the other functions in words.ts as well.